dashboard/src/components/App.js

```diff
@@ -0,0 +1,347 @@
+      console.log("Updated timestamp offset to " + timestampOffset);
+    }
+
+    // updateStatusText('Processing partial data recording with ' + delay + 'ms delay')
+
+    partialDataRecordingContainer.recordings.forEach(function(dataRecording) {
+      if (dataRecordingContainer.getData(dataRecording.dataId).length == 0) {
+        onDataWithNewIdReceived(dataRecording.dataId);
+      }
+      dataRecordingContainer.addDataRecording(dataRecording);
+    });
+    dataRecordingContainer.trim();
+  }
+
+  function onDeviceWithNewIdConnected(device) {
+
+    // update the connected devices array, place the new device first
+    updateConnectedDeviceList((oldConnectedDeviceList) => {
+      var connectionStatus = true;
+
+      oldConnectedDeviceList.forEach(function(oldDevice) {
+        if (oldDevice.id == device.id) {
+          connectionStatus = false;
+        }
+      });
+
+      if (connectionStatus == true) {
+        M.toast({
+          html: device.name + ' connected'
+        });
+        return oldConnectedDeviceList.concat(device);
+      } else {
+        return oldConnectedDeviceList;
+      }
+    });
+  }
+
+  function onDataWithNewIdReceived(id) {
+    // first recording of data with that ID
+    console.log('Received first recording of data with ID: ' + id);
+
+    var ids = dataRecordingContainer.getIds().sort();
+    var optionText = DataRecordingContainer.getReadableId(id);
+
+    updateDataList([]);
+
+    // append available IDs as options
+    ids.forEach(id => {
+      var optionValue = id;
+      var optionText = DataRecordingContainer.getReadableId(id);
+
+      updateDataList((oldDataList) => {
+        return oldDataList.concat({
+          id: optionValue,
+          optionText: optionText
+        });
+      });
+    });
+  }
+
+
+  // HANDLE SELECT-CHANGE
+  function handleDeviceChange(event) {
+
+    setupSocket();
+
+    var selectedDeviceId = event.target.value;
+    selectedDevice = connectedDevices.filter(connectedDevice => connectedDevice.id == selectedDeviceId)[0];
+    console.log('Selected device changed: ' + JSON.stringify(selectedDevice));
+    updateDataRecordingContainer(new DataRecordingContainer());
+  }
+
+  function handleDataChange(event) {
+
+    updateSelectedDataId(event.target.value);
+    console.log('Selected data ID changed: ' + event.target.value);
+    updateRecreateChartPlot(true);
+  }
+
+
+  return (
+    <div className="section">
+    <Logo />
+      <div className="row">
+        <Visualization dataRecordingContainer={dataRecordingContainer} selectedDataId={selectedDataId} />
+        <Row>
+          <Device deviceList={connectedDevices} dataList={dataList} handleDeviceChange={handleDeviceChange} handleDataChange={handleDataChange} />
+          <Info />
+        </Row>
+        {/* <ModalView /> */}
+      </div>
+    </div>
+  );
+}
+
+
+class DataRecordingContainer {
+
+  constructor() {
+    this.dataRecordings = {};
+  }
+
+  get dataRecordings() {
```

Comment:
is it planned for these getters/setters to get additional logic later on? if not remove and just use dataRecordings directly

Reply:
@Steppschuh ?